x-pack/plugins/enterprise_search/public/applications/app_search/components/engine_creation/engine_creation.test.tsx

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React from 'react';
+import { shallow } from 'enzyme';
+
+import '../../../__mocks__/shallow_useeffect.mock';
+import { setMockValues } from '../../../__mocks__';
+
+import { EngineCreation } from './';
+
+describe('EngineCreation', () => {
+  const DEFAULT_VALUES = {
+    name: '',
+    rawName: '',
+    language: 'Universal',
+  };
+
+  describe('default values', () => {
+    beforeEach(() => {
+      jest.clearAllMocks();
+    });
+
+    it('renders', () => {
+      setMockValues(DEFAULT_VALUES);
+      const wrapper = shallow(<EngineCreation />);
+      expect(wrapper.find('[data-test-subj="EngineCreation"]')).toHaveLength(1);
+    });
+
+    it('contains a form', () => {
+      setMockValues(DEFAULT_VALUES);
+      const wrapper = shallow(<EngineCreation />);
+      expect(wrapper.find('[data-test-subj="EngineCreationForm"]')).toHaveLength(1);
+    });
+
+    it('contains a name input', () => {
+      setMockValues(DEFAULT_VALUES);
+      const wrapper = shallow(<EngineCreation />);
+      expect(wrapper.find('[data-test-subj="EngineCreationNameInput"]')).toHaveLength(1);
+    });
+
+    it('contains a language input', () => {
+      setMockValues(DEFAULT_VALUES);
+      const wrapper = shallow(<EngineCreation />);
+      expect(wrapper.find('[data-test-subj="EngineCreationLanguageInput"]')).toHaveLength(1);
+    });
+
+    describe('NewEngineSubmitButton', () => {
+      it('renders', () => {
+        setMockValues(DEFAULT_VALUES);
+        const wrapper = shallow(<EngineCreation />);
+        expect(wrapper.find('[data-test-subj="NewEngineSubmitButton"]')).toHaveLength(1);
+      });
+
+      it('is disabled when name is empty', () => {
+        setMockValues(DEFAULT_VALUES);
+        const wrapper = shallow(<EngineCreation />);
+        expect(wrapper.find('[data-test-subj="NewEngineSubmitButton"]').prop('disabled')).toEqual(
+          true
+        );
+      });
+
+      it('is enabled when name has a value', () => {
+        setMockValues({ ...DEFAULT_VALUES, name: 'test', rawName: 'test' });
+        const wrapper = shallow(<EngineCreation />);
+        expect(wrapper.find('[data-test-subj="NewEngineSubmitButton"]').prop('disabled')).toEqual(
+          false
+        );
+      });
+    });
```

Comment:
We're missing a ton of coverage on the EngineCreation component - try running `sh jest.sh public/applications/app_search/components/engine_creation/` from the Enterprise Search plugin root to take a look at the lines missing coverage.

At a quick glance at the file, here's an example of some functions / branches that I can see that will need to be tested:

```js
it('renders name helptext when the raw name is being sanitized')
it('renders allowed character helptext otherwise')

it('calls setRawName on EngineCreationNameInput change')
// expect(setRawName).toHaveBeenCalled();

it('calls setLanguage on EngineCreationLanguageInput change')
// expect(setLanguage).toHaveBeenCalled();

it('calls submitEngine on form submit')
// ex.
// wrapper.find(form).simulate('submit', { preventDefault: jest.fn() });
// expect(submitEngine).toHaveBeenCalled();

// note: you'll need to call setMockActions to mock jest.fn()s for your logic actions
```

Apologies I couldn't provide more helpful examples, I have to sign off early this evening - don't hesitate to reach out to me tomorrow or ping me if I can help clarify at all